components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/AuthenticationEndpointUtil.java

```diff
@@ -193,5 +195,20 @@ public static String customi18n(ResourceBundle resourceBundle, String key) {
             return i18nBase64(resourceBundle, key);
         }
     }
+
+    /**
+     * Build Tenant Qualified Url.
+     *
+     * @param context Context.
+     * @return Tenant Qualified Url.
+     */
+    public static String buildTenantQualifiedUrl(String context) {
+
+        try {
+            return ServiceURLBuilder.create().addPath(context).build().getAbsoluteInternalURL();
+        } catch (URLBuilderException e) {
+            return null;
```

Comment:
Wouldn't it be better to log the error?

Reply:
Wouldn't it be better if we throw a wrapped exception and handle at the consuming places?

Reply:
Updated the PR as discussed.